src/runtime/contrib/arm_compute_lib/acl_runtime.cc

```diff
@@ -0,0 +1,309 @@
+
+    auto function = std::make_shared<arm_compute::NEConvolutionLayer>(mm);
+    function->configure(&layer->inputs[0], &layer->const_inputs[0],
+                        layer->const_inputs.size() > 1 ? &layer->const_inputs[1] : nullptr,
+                        &layer->outputs[0], pad_stride_info, arm_compute::WeightsInfo(),
+                        dilation_2d, act_info);
+    layer->function = function;
+  }
+
+  /*!
+   * \brief Create a pooling layer.
+   *
+   * \note Currently only maxpool is supported.
+   *
+   * \param layer The ACL layer to build. Containing inputs, outputs and the ACL function.
+   * \param node The JSON representation of the operator.
+   */
+  static void CreatePoolingLayer(CachedLayer* layer, const JSONGraphNode& node) {
+    std::vector<std::string> padding = node.GetAttr<std::vector<std::string>>("padding");
+    std::vector<std::string> strides = node.GetAttr<std::vector<std::string>>("strides");
+    arm_compute::PadStrideInfo pad_stride_info = ToACLPadStride(padding, strides);
+
+    auto attr_pool_size = node.GetAttr<std::vector<std::string>>("pool_size");
+    int pool_size_h = std::stoi(attr_pool_size[0]);
+    int pool_size_w = std::stoi(attr_pool_size[1]);
+
+    arm_compute::PoolingType pool_type;
+    if (node.GetOpName() == "nn.max_pool2d") {
+      pool_type = arm_compute::PoolingType::MAX;
+    } else {
+      LOG(FATAL) << "Pooling type not supported";
+    }
+
+    arm_compute::PoolingLayerInfo pool_info =
+        arm_compute::PoolingLayerInfo(pool_type, arm_compute::Size2D(pool_size_h, pool_size_w),
+                                      arm_compute::DataLayout::NHWC, pad_stride_info);
+
+    layer->outputs.push_back(MakeOutputTensor(node.GetOpShape()[0]));
+
+    auto function = std::make_shared<arm_compute::NEPoolingLayer>();
+    function->configure(&layer->inputs[0], &layer->outputs[0], pool_info);
+    layer->function = function;
+  }
+
+  /*!
+   * \brief Create a reshape layer.
+   *
+   * \param layer The ACL layer to build. Containing inputs, outputs and the ACL function.
+   * \param node The JSON representation of the operator.
+   */
+  static void CreateReshapeLayer(CachedLayer* layer, const JSONGraphNode& node) {
+    layer->outputs.push_back(MakeOutputTensor(node.GetOpShape()[0]));
+    auto function = std::make_shared<arm_compute::NEReshapeLayer>();
+    function->configure(&layer->inputs[0], &layer->outputs[0]);
+    layer->function = function;
+  }
+
+  /*! \brief Allow ACL functions to request auxiliary memory from TVM. */
+  arm_compute_lib::ACLAllocator allocator_;
+  /*!
+   * \brief The network layers represented by acl functions.
+   * \note Currently only supports a single layer.
+   */
+  CachedLayer layer_;
+#else
+  void Run() override {
+    LOG(FATAL) << "Cannot call run on Arm Compute Library module without runtime enabled. "
+               << "Please build with USE_ARM_COMPUTE_LIB_GRAPH_RUNTIME.";
+  }
+
+  void BuildEngine() {
+    // Do nothing.
```

Comment:
Maybe we could have a warning `LOG(WARNING)` to mention the engine is actually not initialized.